hftbacktest/src/types.rs

```diff
@@ -702,7 +702,7 @@ impl Encode for Order {
 }
 
 /// An asynchronous request to [`Connector`](`crate::connector::Connector`).
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Encode, Decode)]
```

Comment:
Why do we need to do serialization? Can't we do zero-copy sends with iceoryx2 over shmem?

Reply:
Seems like we can: https://github.com/nkaz001/hftbacktest/commit/3e259be0349e50fe6ad580f6972bcfa78119b3dc

Reply:
I thought it wasn't possible to use it directly because the `Box` in `Order` prevents it from being a POD type.
I'm curious how `iceoryx2` handles de/serialization in this case. If `iceoryx2` manages this properly, it may eliminate the need for an additional de/serialization layer. I will take a closer look at it.

Reply:
That does pose a challenge, but I think at the worst we could figure a way to allocate the trait object in the shmem segment used by iceoryx2 and the pointer should work across all consumers.

Reply:
It cannot be directly processed by iceoryx due to presumably the Box. A runtime error(invalid pointer) occurred.

Reply:
We can probably figure out a way to have the order data allocated inside the shmem and reconstruct the fat pointer on the consumer side by doing a `type_id` sort of thing to support backtesting. I can take a look if you'd like.

Reply:
Yes, but I think that falls within the scope of iceoryx. It would be better for iceoryx to provide a trait for custom de/serialization since we don't need to actually transfer the `Box`. Until the necessary features are ready, it's simpler to stick with bincode de/serialization and refactor later.

Reply:
Even if/when iceoryx provides serialization support it will still have to do a complete copy, which will be orders of magnitude slower than allocating in and reading directly from shmem.

I think the only way to properly handle this in the context of iceoryx is for the type containing `q` to be aware that it can be relocated to a different address space, or to refactor so that `Order` doesn't contain a `Box` at all. Maybe iceoryx can provide some APIs in the future to help with that, but I can't think of any way it would be able to handle `Order` as it's written today without involving copies.

Reply:
In any case, `Order` and `Event` are copied unless they are allocated in shared memory initially. We don't need to be concerned with the contents of the `Box`, as I mentioned earlier, it's null in a live bot; we just need to manage its creation and deletion. Ideally, the best approach would be to use the allocator feature with a shared memory allocator. However, this feature is still unstable in Rust.

Reply:
> Ideally, the best approach would be to use the allocator feature with a shared memory allocator. However, this feature is still unstable in Rust.

FWIW iceoryx provides equivalent versions of these APIs. E.g., see https://github.com/eclipse-iceoryx/iceoryx2/blob/main/iceoryx2-cal/src/shm_allocator/pool_allocator.rs#L56.